### DISCUSSION TITLE
Flutter UI

### DISCUSSION THREAD

**Discussion by Khaled-BH** [TDRA-ae/aegov-dls#43] [General]
Hey, I'm Planning to build most of components on Flutter,
Any suggestion ?

![image](https://github.com/TDRA-ae/aegov-dls/assets/50563995/799968dc-d295-48fb-974a-d999df01c514)

**Comment by moiz-conceptlz** [comment]
Using the design system to build a mobile app is entirely possible; however, the current version of the design system is not completely optimized to provide every component that may be required for a mobile app.

Using Flutter is great. However, if you simply extend https://docs.flutter.dev/ui/widgets/material and aim to style it, by changing colours, it will miss the details that are present in the design system.

For example, in your screenshot as well - you will notice that the border-radius is incorrect, the soft button has a shadow and dropdown paddings compared to the text size is incorrect.  Therefore, to effectively use the design system - each component will need to be created as a native Flutter component. This is only a suggestion, and still open to discussion on the benefits of extending an existing competent library and styling it v/s building a library from scratch.

**Reply by Khaled-BH** [reply]
Yes, You're right.

I can manage this to be with UAE Design system guideline. 
The goal of this project is to create ready-made templates that anyone can use.

**Reply by Khaled-BH** [reply]
And also , on these widgets I didn't used any external packages.
I will make it native Flutter component.

**Reply by moiz-conceptlz** [reply]
Great @Khaled-BH - all the best. Good initiative.
